### ISSUE TITLE
Overriding Angular's email validator

### ISSUE THREAD

**Issue by sebastian-zarzycki-es** [rsaenen/ng4-validators#5]
It seems that newer versions of Angular also have email validator. This library also offers email validator that seems to be more tight and restrictive (which is good). I wonder how it works, though, as the selector seems to be exactly the same. Does Angular run both validators under the same name, or is it overridden somehow (because declared later?)

**Comment by rsaenen** [comment]
Angular directive selector:
```typescript
selector: '[email][formControlName],[email][formControl],[email][ngModel]'
 ```

Custom ng4-validators directive selector:
```typescript
selector: '[email][formControlName],[email][formControl],[email][ngModel]',
 ```

So it both has the same selector, I don't check if the custom validator override angular validator since angular 5 release.

I'll check this behavior, but I guess it's better to rename the custom directive selector to avoid issue.

**Comment by sebastian-zarzycki-es** [comment]
I would think so, yes, it's probably better to rename. 

But, apart from that, I'm a little bit confused as to how this override works, and whether it allows the previous validator run in parallel, or not.

**Comment by rsaenen** [comment]
It think there's no override but just javascript runs both and the result is given from the last executed validator.

To override default validator, I guess the custom has to implement an interface like EmailValidatorInterface but without I don't guess that is possible.

**Comment by sebastian-zarzycki-es** [comment]
Yeah, the "last executed" part interests me. What is the order? Is it always Angular first, then whatever's declared later? Just curious. Sorry, this probably isn't a bug request anymore. Just worth changing that email validator to avoid confusion.

**Comment by rsaenen** [comment]
I'am unable to publish a new version for now, see [angular/cli#9549](https://github.com/angular/angular-cli/issues/9549).
I won't publish without run tests.

**Comment by rsaenen** [comment]
Released with 5.0.1
